app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb

```diff
@@ -1,42 +1,48 @@
-class ManageIQ::Providers::Nuage::NetworkManager::EventCatcher::MessagingHandler < Qpid::Proton::Handler::MessagingHandler
+class ManageIQ::Providers::Nuage::NetworkManager::EventCatcher::MessagingHandler < Qpid::Proton::MessagingHandler
   def initialize(options = {})
     require 'qpid_proton'
 
     super()
     @options = options
 
-    @topics = @options.delete(:topics)
-    @test_connection = @options.delete(:test_connection)
+    @topics                = @options.delete(:topics)
+    @test_connection       = @options.delete(:test_connection)
     @message_handler_block = @options.delete(:message_handler_block)
+    @url                   = @options.delete(:url)
+    @timeout               = @options.delete(:amqp_connect_timeout) || 5.seconds
   end
 
-  def on_start(event)
-    @conn = event.container.connect(@options)
+  def on_container_start(container)
+    Timeout.timeout(@timeout) { @conn = container.connect(@url, @options) }
```

Comment:
You are using the `Timeout` here because Qpid does not provide such mechanism on its own?

Reply:
Yes, the only timeout in whole qpid_proton ruby binding that I could find was:

```
 :idle_timeout (Numeric) — Seconds before closing an idle connection
```

which is not what we need here. So when I intentionaly input following AMQP urls when adding new provider:

```
google.com
yahoo.com
bbc.com
```

the UI got blocked for 3 minutes :D